lib/tasks/pwpush.rake

```diff
@@ -1,19 +1,27 @@
 # frozen_string_literal: true
 
 # A task that should be run periodically (daily/weekly) to iterate through all unexpired
-# pushes and run .validate! which will determine if a push has expired or not.
+# pushes and run .check_limits which will determine if a push has expired or not.
 #
-# Note: .validate! is also run on each attempt to view an unexpired secret URL so this task is
+# Note: .check_limits is also run on each attempt to view an unexpired secret URL so this task is
 # a preemptive measure to expire pushes periodically.  It saves some CPU and DB calls
 # on live requests.
 #
-desc "Run through, validate and conditionally expire passwords."
+desc "Run through, validate and conditionally expire pushes."
 task daily_expiration: :environment do
   counter = 0
   expiration_count = 0
 
   puts "--> Starting daily expiration on #{Time.zone.now}"
 
+  Push.where(expired: false).find_each do |push|
+    counter += 1
+    push.check_limits
+    expiration_count += 1 if push.expired
+  end
+
+  puts "  -> Finished validating #{counter} unexpired pushes.  #{expiration_count} total pushes expired..."
+
   Password.where(expired: false).find_each do |push|
     counter += 1
     push.validate!
```

Comment:
I haven't removed tasks and background jobs for old push records. Tasks using old push records will be removed when related tables are removed.